### PR TITLE
Fixes Partner targeting and Acupressure/Ally Switch interaction

### LINKED ISSUE #4565: Ally Switch + Acupressure should fail when new target becomes the user
### Description

The following test isn't behaving as it does in at least Gen 5.
```
// Verified on Showdown, even though Bulbapedia says otherwise.
DOUBLE_BATTLE_TEST("Acupressure works after ally used Ally Switch")
```
Unfortunetly, "works on Showdown" isn't gonna cut it anymore, because Smogon isn't going to test all non-viable edge cases.

### Expansion: Target left vs right (user Wynaut)
mGBA_7l5BDRqH5x mGBA_oxzPQ0qbBR

### Gen 5: Target left vs right (user Doduo)
melonDS_hXe0cvYdaC melonDS_3RN9B4KFVg

### Version

1.8.3 (Latest release)

### Upcoming/master Version

ebdc9ffc39eb6443116bef155fb4840396f67ad3

### Discord contact info

AsparagusEduardo

### LINKED ISSUE #5154: [BUG] Move that targets partner is redirected if partner dies before that
### Description

I'm not sure if this is a new behavior or it was always like that but in vanilla games the move that was supposed to target partner will fail if the target partner faints.

On upcoming / master the move redirects opponent.  
```
DOUBLE_BATTLE_TEST("Move should fail if it targets partner but it faints before that")
{
    GIVEN {
        PLAYER(SPECIES_WOBBUFFET);
        PLAYER(SPECIES_WOBBUFFET) { HP(1); }
        OPPONENT(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(opponentLeft, MOVE_TACKLE, target: playerRight); MOVE(playerLeft, MOVE_TACKLE, target: playerRight); }
    } SCENE {
        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponentLeft);
        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerLeft);
    }
}
```

### Version

1.9.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #4565
Fixes #5154

It is a bit of a scary change but all tests pass and I don't know what else could break

### CODE REVIEW

File: test/battle/move_effect/acupressure.c [hunk 1]
```diff
@@ -8,29 +8,8 @@ TO_DO_BATTLE_TEST("Acupressure fails on the ally if all of its stats are maximiz
 TO_DO_BATTLE_TEST("Acupressure works on the user if it's behind a Substitute (Gen5+)");
 TO_DO_BATTLE_TEST("Acupressure fails on its ally if it's behind a Substitute");
 
-DOUBLE_BATTLE_TEST("Acupressure works on the ally if the user targeted itself but switched positions via Ally Switch")
```
[the remark this answers is not in the document]
Comment (AlexOn1ine) [reply test/battle/move_effect/acupressure.c:11]: The test was wrong. The known failing test is the correct one and afaik it never changed in any generation

**Comment by AlexOn1ine** [comment]
putting on draft because I thought of more tests to write

**Comment by AlexOn1ine** [comment]
Added more tests. Ready for review again.

File: test/battle/move.c [hunk 2]
```diff
@@ -222,3 +222,71 @@ SINGLE_BATTLE_TEST("Critical hits ignore negative stat stages", s16 damage)
...
+    } WHEN {
+        TURN {
+            MOVE(playerLeft, MOVE_HYPER_VOICE);
+            MOVE(playerRight, MOVE_TACKLE, target: opponentLeft);
+            SEND_OUT(opponentLeft, 2);
+            SEND_OUT(opponentRight, 3);
+        }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_HYPER_VOICE, playerLeft);
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerLeft);
```
Comment (Pawkkie) [inline test/battle/move.c:274]: I think this should be `playerRight`

**Review by Pawkkie — CHANGES_REQUESTED** [review]
I agree this is a bit spooky, I'm willing to put my trust in the tests and I think the changes look sound as is otherwise. Happy to wait for an additional reviewer if you like :)

**Comment by Pawkkie** [comment]
> Fwiw I don't think this will break anything but if you would like a second opinion I could ask someone else to take a look.

No I'm also pretty confident it's good to go barring what I think is the test typo, if you're also comfortable with it I don't think we need to hassle others :)

**Comment by AlexOn1ine** [comment]
> > Fwiw I don't think this will break anything but if you would like a second opinion I could ask someone else to take a look.
> 
> No I'm also pretty confident it's good to go barring what I think is the test typo, if you're also comfortable with it I don't think we need to hassle others :)

as psf would say, merge that bitch

**Comment by Pawkkie** [comment]
> as psf would say, merge that bitch

First though was the test I commented on a typo though or my error lol

File: test/battle/move.c (hunk 2 above)
Comment (AlexOn1ine) [inline test/battle/move.c:274]:
```suggestion
        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerRight);
```

**Review by Pawkkie — APPROVED** [review]
